### PR TITLE
Add conditional xaxis formatting for parameter plots

### PR DESCRIPTION
**Pull request by xjules**
**Issue**
Resolves https://github.com/equinor/ert/issues/12011

**Approach**
Introduces ConditionalFormatter class

(Screenshot of new behavior in GUI if applicable)
image

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'just rapid-tests'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Add backport label to latest release (format: 'backport release-branch-name')

### CODE REVIEW

File: src/ert/gui/tools/plot/plottery/plots/plot_tools.py [hunk 1]
```diff
@@ -11,6 +13,44 @@
...
+        high: float = 1e4,
+        precision: float = 0,
+    ) -> None:
+        self.low = low
+        self.high = high
+        self.precision = precision
+
+    def __call__(self, x: float, pos: int | None = None) -> str:
+        if x == 0 or (self.low <= abs(x) < self.high):
+            return f"{x:.6g}"
```
Comment (andreas-el) [inline src/ert/gui/tools/plot/plottery/plots/plot_tools.py:48]: I assume that float can also be `INF`, `-INF` or `NaN` ? If so, this might cause some issues.

File: tests/ert/unit_tests/gui/plottery/test_plot_style.py [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+    assert fmt(0) == "0"
+    assert fmt(1e-3) == "0.001"
+    assert fmt(123) == "123"
+    assert fmt(1e3) == "1000"
+
+    assert fmt(2e-2) == "0.02"
+    assert fmt(2e-6) == "2e-6"
+    assert fmt(-2e-6) == "-2e-6"
+    assert fmt(2e5) == "2e5"
+    assert fmt(-3e6) == "-3e6"
```
Comment (andreas-el) [inline tests/ert/unit_tests/gui/plottery/test_plot_style.py:24]: Should also test `1e4` which is the boundary you set.
In general we should test for each boundary: `0`, `1e-3` and `1e4` both regular and negated values.
- Exact
- Slightly above
- Slightly below

File: src/ert/gui/tools/plot/plottery/plots/plot_tools.py (hunk 1 above)
Comment (xjules) [reply src/ert/gui/tools/plot/plottery/plots/plot_tools.py:48]: Not sure, as I guess this would fail by some matplotlib validation first. I can test it nonetheless.

File: src/ert/gui/tools/plot/plottery/plots/plot_tools.py [hunk 3]
```diff
@@ -11,6 +13,44 @@
...
+        self,
+        low: float = 1e-3,
+        high: float = 1e4,
+        precision: float = 0,
+    ) -> None:
+        self.low = low
+        self.high = high
+        self.precision = precision
+
+    def __call__(self, x: float, pos: int | None = None) -> str:
```
Comment (andreas-el) [inline src/ert/gui/tools/plot/plottery/plots/plot_tools.py:42]:
```
if x in [math.inf, -math.inf, math.nan]:
    return f"{x:.6g}"
```
Perhaps this will work?

File: src/ert/gui/tools/plot/plottery/plots/plot_tools.py (hunk 1 above)
Comment (andreas-el) [reply src/ert/gui/tools/plot/plottery/plots/plot_tools.py:48]: Given the float special handling, you could increase the test with more examples;
```
    assert fmt(0) == "0"
    assert fmt(-0) == "0"
    assert fmt(0.001) == "0.001"
    assert fmt(-0.001) == "-0.001"
    assert fmt(0.0001) == "1e-4"
    assert fmt(-0.0001) == "-1e-4"
    assert fmt(123) == "123"
    assert fmt(1e3) == "1000"
    assert fmt(1e4) == "1e4"
    assert fmt(2e-2) == "0.02"
    assert fmt(2e-6) == "2e-6"
    assert fmt(-2e-6) == "-2e-6"
    assert fmt(2e5) == "2e5"
    assert fmt(-3e6) == "-3e6"
    assert fmt(10000) == "1e4"
    assert fmt(-10000) == "-1e4"
    assert fmt(9999) == "9999"
    assert fmt(-9999) == "-9999"

    import math
    assert fmt(math.nan) == "nan"
    assert fmt(math.inf) == "inf"
    assert fmt(-math.inf) == "-inf"
```
